packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -266,3 +268,78 @@ class ScrollController extends ChangeNotifier {
     }
   }
 }
+
+/// A [ScrollController] whose `initialScrollOffset` tracks its most recently
+/// updated [ScrollPosition].
+///
+/// This class can be used to synchronize the scroll offset of two or more
+/// lazily created scroll views that share a single [TrackingScrollController].
+/// It tracks the most recently updated scroll position and reports it as its
+/// `initialScrollOffset`.
+///
+/// ## Sample code
+///
+/// In this example each [PageView] page contains a [ListView] and all three
+/// [ListView]'s share a [TrackingController]. The scroll offsets of all three
+/// list views will track each other, to the extent that's possible given the
+/// different list lengths.
+///
+/// ```dart
+/// new PageView(
+///   children: <Widget>[
+///     new ListView(
+///       controller: _trackingScrollController,
+///       children: new List<Widget>.generate(100, (int i) => new Text('page 0 item $i')).toList(),
+///     ),
+///    new ListView(
+///      controller: _trackingScrollController,
+///      children: new List<Widget>.generate(200, (int i) => new Text('page 1 item $i')).toList(),
+///    ),
+///    new ListView(
+///      controller: _trackingScrollController,
+///      children: new List<Widget>.generate(300, (int i) => new Text('page 2 item $i')).toList(),
+///     ),
+///   ],
+/// )
+/// ```
+///
+/// In this example the `_trackingController` would have been created by the
+/// stateful widget that built the widget tree.
+class TrackingScrollController extends ScrollController {
+  Map<ScrollPosition, VoidCallback> _positionToListener = <ScrollPosition, VoidCallback>{};
+  ScrollPosition _lastUpdated;
+
+  /// The last [ScrollPosition] to change. Returns null if there aren't any
+  /// attached scroll positions or there hasn't been any scrolling yet.
+  ScrollPosition get mostRecentlyUpdatedPosition => _lastUpdated;
+
+  /// Returns the scroll offset of the [mostRecentlyUpdatedPosition] or 0.0.
+  @override
+  double get initialScrollOffset => _lastUpdated?.pixels ?? 0.0;
```

Comment:
Rather than returning 0.0 by default, return super.initialScrollOffset, and then have TrackingScrollController take an initialScrollOffset which it passes to the parent. That way, you can still set an initial scroll offset.